enumerable_methods.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    my_each do |element|
+      selected_array << element if yield(element)
+    end
+    selected_array
+  end
+
+  def my_all?(_para = nil)
+    result = true
+    if block_given?
+      my_each do |element|
+        result = false unless yield(element)
+      end
+    else
+      my_each do |element|
+        result = false unless element
+      end
+    end
+    result
+  end
+
+  def my_any?(para = nil)
+    return my_any?(para) if block_given? && !para.nil?
+
+    if block_given?
+      to_a.my_each do |element|
+        return true if yield element
+
+        false
+      end
+    elsif para
+      to_a.my_each do |element|
+        return true if element == para
+      end
+    elsif para.nil?
+      to_a.my_each do |element|
+        return true if element
+      end
+    end
+    false
+  end
+
+  def my_none?(para = nil)
+    return my_none?(para) if block_given? && !para.nil?
+
+    if block_given?
+      to_a.my_each do |element|
+        return false if yield element
+
+        false
+      end
+    elsif para
+      to_a.my_each do |element|
+        return false if element == para
+      end
+    elsif para.nil?
+      to_a.my_each do |element|
+        return false if element
+      end
+    end
+    true
+  end
+
+  def my_count(para = nil)
+    count = 0
+    if block_given?
+      my_each do |element|
+        count += 1 if yield(element)
+      end
+    elsif para
+      my_each do |element|
+        count += 1 if element == para
+      end
+    else
+      count = size
+    end
+    count
+  end
+
+  def my_map(prc = nil)
+    arr = []
+    if prc
+      my_each do |elem|
+        arr << prc.call(elem)
+      end
+    else
+      my_each do |elem|
+        arr << yield(elem)
+      end
+    end
+    arr
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] returns an Enumerator if no block is given
```ruby
[2,4,7,11].my_map ==  [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map
```
- Yours is retrieving a LocalJumpError
![image](https://user-images.githubusercontent.com/52209067/80105390-c8645180-8535-11ea-9292-a9e6748e367c.png)

- Real map
![image](https://user-images.githubusercontent.com/52209067/77697646-dc3b8880-6f74-11ea-9277-4ed4a2b3af4c.png)